### PR TITLE
bread-crumbs

### PR DESCRIPTION
**Pull request by serhii-lisniakov**

### CODE REVIEW

**Review by v-skochko — APPROVED** [review]
Done

File: src/app/header/bread-crumbs/bread-crumbs.component.ts [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
+  constructor(private breadCrumbsService: BreadCrumbsService) { }
 
   ngOnInit(): void {
+    this.breadCrumbsService.routeSubscribe();
+    this.breadCrumbsService.crumbs.pipe(
+      takeUntil(this.destroy$)
+    ).subscribe(crumbs => this.crumbs = crumbs);
+    this.breadCrumbsService.title.pipe(
+      takeUntil(this.destroy$)
+    ).subscribe(title => this.title = title.charAt(0).toUpperCase() + title.slice(1));
```
Comment (Vsumskyi) [inline src/app/header/bread-crumbs/bread-crumbs.component.ts]: Separate it, please

File: src/app/services/bread-crumbs.service.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+export class BreadCrumbsService {
+  public crumbs = new BehaviorSubject([]);
+  public title = new BehaviorSubject('');
+  constructor(private router: Router) { }
+
+  routeSubscribe(): void {
+    this.router.events.subscribe(event => {
+      if (event instanceof ChildActivationStart) {
+        this.crumbs.next(event.snapshot.data.crumbs);
+      }
```
Comment (Vsumskyi) [inline src/app/services/bread-crumbs.service.ts]: Why are you doing subscribe in service, in my opinion it is a bad way

File: src/app/services/bread-crumbs.service.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BreadCrumbsService {
+  public crumbs = new BehaviorSubject([]);
+  public title = new BehaviorSubject('');
+  constructor(private router: Router) { }
+
+  routeSubscribe(): void {
```
Comment (Vsumskyi) [inline src/app/services/bread-crumbs.service.ts]: private or public?

File: src/app/body/product/product.component.ts [hunk 4]
```diff
@@ -14,18 +15,21 @@ export class ProductComponent implements OnInit {
...
               private route: ActivatedRoute,
               private router: Router) { }
 
   ngOnInit(): void {
     this.getProduct();
     this.subscribeToRouteChanges();
+    this.breadCrumbsService.title.next(this.product.title);
   }
 
   private subscribeToRouteChanges(): void {
```
Comment (Vsumskyi) [inline src/app/body/product/product.component.ts:36]: Problaly, you should unsubscrive

Comment (serhii-lisniakov) [reply src/app/body/product/product.component.ts:36]: fixed

File: src/app/services/bread-crumbs.service.ts (hunk 3 above)
Comment (serhii-lisniakov) [reply src/app/services/bread-crumbs.service.ts]: fixed

File: src/app/services/bread-crumbs.service.ts (hunk 2 above)
Comment (serhii-lisniakov) [reply src/app/services/bread-crumbs.service.ts]: fixed

File: src/app/header/bread-crumbs/bread-crumbs.component.ts (hunk 1 above)
Comment (serhii-lisniakov) [reply src/app/header/bread-crumbs/bread-crumbs.component.ts]: fixed

**Review by Vsumskyi — APPROVED** [review]
Aproved

**Pull request merged**